### DISCUSSION TITLE
How do I get my device's exact model code?

### DISCUSSION THREAD

**Discussion by ggonmar** [Tasshack/dreame-vacuum#260] [Q&A] [answered]
Hi,

I reached this repo searching for some way to get my vacuum working on Home Assistant: I have a Xiaomi Mi Robot Vacuum-Mop 2, which seems to be supported given what is listed on the README. 

However, when I try to add the integration, I get "No supported devices found in this Xiaomi Miio cloud account on selected country." - I tried on all servers just in case, always same result. 
I'd greatly appreciate some help!

Thank you

**Comment by Tasshack** [comment, marked as answer]
Your device is probably not the Mop 2 model which made was by the Dreame.
You can get device model number using this https://github.com/PiotrMachowski/Xiaomi-cloud-tokens-extractor.

**Comment by wqrwefdgfv** [comment]
Hi! What about Dreame Trouver Robot Vacuum M1 RPM1GA?  Is it works with this integration?

**Reply by Tasshack** [reply]
It does not have a lidar tower or a vslam camera like the rest of the supported devices so I don't know how it's map compatible with the current implementation. I need access to a one to make sure it will work with the integration or make necessary changes to make it working.

**Reply by wqrwefdgfv** [reply]
It has "Solid-Fixed Laser Navigation (SoF LiDAR)"

**Reply by Tasshack** [reply]
Yes I know that but I don't know how it stores the map.

**Reply by Tasshack** [reply]
@wqrwefdgfv https://github.com/Tasshack/dreame-vacuum/issues/607
